crates/engine/tree/src/tree/mod.rs

```diff
@@ -605,12 +608,10 @@ where
                         error!("could not find persisted block with hash {last_persisted_block_hash} in memory");
                     }
                 }
-                Err(TryRecvError::Closed) => {
-                    panic!("persistence task has been terminated");
-                }
-                _ => {}
+                Err(err) => return Err(err),
```

Comment:
we need to handle the `TryRecvError::Empty` case here and assign the rx back to `.persistence_state` otherwise we're only trying to receive it once and dropping the receiver if there's no response yet

Reply:
sure, sorry missed that part, done here https://github.com/paradigmxyz/reth/pull/10133/commits/dde3f1c13570ed2a4fb5f0c6b92581f341853561